packages/solo/req.go

```diff
@@ -396,7 +404,37 @@ func (ch *Chain) EstimateGasOffLedger(req *CallParams, keyPair *cryptolib.KeyPai
 	}
 	r := req.NewRequestOffLedger(ch.ChainID, keyPair)
 	res := ch.estimateGas(r)
-	return res.Receipt.GasBurned, res.Receipt.GasFeeCharged, res.Receipt.Error()
+
+	return res.Receipt.GasBurned, res.Receipt.GasFeeCharged, res.Receipt.Error.AsGoError()
+}
+
+// ErrorMessageResolver has the signature of VMErrorMessageResolver to provide a way to resolve the error format
+func (ch *Chain) ErrorMessageResolver() func(vmError *iscp.UnresolvedVMError) (string, error) {
```

Comment:
```suggestion
func (ch *Chain) ErrorMessageResolver(vmError *iscp.UnresolvedVMError) (string, error) {
```
wouldn't this work? why do we need to return a new function?